File: GettingStartedWithKIF/GettingStartedWithKIF/ViewController.m
//
//  ViewController.m
//  GettingStartedWithKIF
//
//  Created by Adam Johns on 1/31/15.
//  Copyright (c) 2015 Adam Johns. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@property (nonatomic) int count;
@property (weak, nonatomic) IBOutlet UILabel *countLabel;
- (IBAction)addTapped:(id)sender;
- (IBAction)subtractTapped:(id)sender;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}

- (IBAction)addTapped:(id)sender {
    self.count++;
    [self updateCountLabel];
}

- (IBAction)subtractTapped:(id)sender {
    self.count--;
    [self updateCountLabel];
}

- (void)updateCountLabel {
    self.countLabel.text = [NSString stringWithFormat:@"%d", self.count];
}

@end

File: 2015.04.18 - UITextFieldTesting/UITextFieldTesting/ViewController.swift
//
//  ViewController.swift
//  UITextFieldTesting
//
//  Created by Adam Johns on 4/18/15.
//  Copyright (c) 2015 Adam Johns. All rights reserved.
//

import UIKit

let maxNumCharacters = 5

class ViewController: UIViewController {
    @IBOutlet weak var textField: UITextField!
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }
}

extension ViewController: UITextFieldDelegate {
    func textField(textField: UITextField, shouldChangeCharactersInRange range: NSRange, replacementString string: String) -> Bool {
        let newLength = count(textField.text) + count(string) - range.length
        return newLength <= maxNumCharacters
    }
}
